src/foolscap/negotiate.py

```diff
@@ -252,15 +252,13 @@ def parseLines(self, header):
         return block
 
     def sendBlock(self, block):
-        keys = block.keys()
-        keys.sort()
-        for k in keys:
+        for k in sorted(block.keys()):
             self.transport.write("%s: %s\r\n" % (k.lower(), block[k]))
         self.transport.write("\r\n") # end block
 
     def debug_doTimer(self, name, timeout, call, *args):
-        if (self._test_options.has_key("debug_slow_%s" % name) and
-            not self.debugTimers.has_key(name)):
+        if ("debug_slow_%s" % name in self._test_options and
+            name not in self.debugTimers):
             self.log("debug_doTimer(%s)" % name)
```

Comment:
This is missing some parens I think, isn't it? or at least, *could* use some more because I can't remember if `%` or `in` is higher-precedence ;)

Reply:
It isn't but will add parens anyway.